common/error.go

```diff
@@ -59,12 +67,34 @@ var errorString = map[uintptr]string{
 	0x032: "SPXERR_CANCELED",
 }
 
-func NewCarbonError(code uintptr) CarbonError {
+func NewCarbonError(errorHandle uintptr) CarbonError {
 	var error CarbonError
-	error.Code = code
+	error.Code = GetErrorCode(SPXHandle(errorHandle))
+	error.Message = GetErrorMessage(SPXHandle(errorHandle))
+	// When the message is empty, construct the error message using the errorHandle value directly.
+	if (error.Message == "") {
+		codeAsHexString := fmt.Sprintf("0x%0x", error.Code)
+		error.Message = "Exception with an error code: " + codeAsHexString + " (" + errorString[error.Code] + ")"
+	}
 	return error
 }
 
 func (e CarbonError) Error() string {
-	return errorString[e.Code]
+	return e.Message
 }
+
+func GetErrorCode(errorHandle SPXHandle) int {
```

Comment:
We probably want these functions to be private, right? (i.e call them getErrorCode and getErrorMessage)